setuptools_rust/setuptools_ext.py

```diff
@@ -96,7 +97,15 @@ def make_distribution(self) -> None:
                 #
                 # https://doc.rust-lang.org/cargo/commands/cargo-build.html#manifest-options
                 cargo_manifest_args: Set[str] = set()
+                env: Optional[Env] = None
                 for ext in self.distribution.rust_extensions:
+                    if env is not None:
+                        if ext.env != env:
+                            raise ValueError(
+                                "For vendoring, all extensions must have the same environment variables"
```

Comment:
It might be nice to improve this message with something like "Extension('foo') has explicit env passed as an option, but other extensions had different environment (or no environment)"

Not sure that is exactly the right wording, but it might help users understand what they did wrong? 🤔